run_tests.sh

```diff
@@ -0,0 +1,7 @@
st,RDFDatetimeTest,FilehashTest,XoredSearchingTest,GRRFuseTest,ClientCommsTest,FilterTests,AFF4SparseImageTest,ByteSizeTest,OSXEnumerateRunningServicesTest,TimelineTest,TestTimelines,TestPackedVersionedCollection,VFSDirectoryTest,FindTest,RekallTests,RDFIntegerTest,TestExecutePython,MethodTest,EmailPluginTest,PackedVersionedCollectionCompactorTest,OSXMemoryTest,HashFileStoreTest,ArtifactFlowTest,AFF4ObjectLabelsListTest,TestExports,AuthTest,ConfigLibTest,GrepTests,GrrKbTest,TestArtifactCollectors,TriggersTest,UsersTest,SshdConfigTest,TestLinuxMemory,DurationTest,ClientURNTests,SqliteDataStoreTest,DataStoreTest,AFF4ObjectLabelTest,TestCheckFlows,TDBDataStoreBenchmarks,ThreadPoolTest,AFF4GRRTest,ChromeHistoryTest,GetClientStatsActionTest,TestSearchFileContent,TestRegistryFinderFlow,AFF4LabelsIndexTest,ListProcessesTest,Ipv6UtilsTest,RDFStructsTest,BuildTests,TestPlistFlows,FileStoreHashTest,ApiConfigRendererTest,IEHistoryTest,RunningStatsTest,AdministrativeFlowTests,YamlConfigTest,RDFValueBaseTest,RDFBytesTest,FileExportPluginTest,TestCopyPathToFile,SqliteDataStoreCSVBenchmarks,RDFValueBenchmark,TDBDataStoreTest,TestLinuxMemoryAnalysis,StoreTests,DataStoreBenchmarks,HintsTests,RekallResponseConverterTest,WindowsRegistryParserTest,ClientUtilsTest,GRRTempFileTestDirectory,EmptyActionTest,FakeDataStoreBenchmarks,AverageMicroBenchmarks,TestListVolumeShadowCopies,WindowsPersistenceMechanismsParserTest,HashDigestTest,WMIParserTest,CollectionExportPluginTest,StatsTests,MasterTest,ApiAff4RendererTest,SendEmailTests,TestClientInterrogate,TestExportHuntResultsFilesAsArchive,RDFURNTest,LinuxSoftwareParserTest,ArtifactParserTest,TestFilesystem,HuntTest,BatchConverterTest,TestArtifactCollectorsInteractions,OSXQuarantineTest,CollectorTest,CollectionFilesExportPluginTest,StandardHuntTest,UserTests,TestSearchFileContentWithFixture,TestCollections,ArtifactParserTests,DarwinPersistenceMechanismsParserTest,RDFValueCollectionApiObjectRendererTest,WebHistoryFlowTest,HTTPDataStoreTest,EmbeddedRDFValueTest,DataAgnosticExportConverterTest,GRRSeleniumTest,SqliteDataStoreBenchmarks,TestSystemRootSystemDriveFallbackFlow,OutputpluginsTest,ConfigFileTest,FlowTest,RegistryFlowTest,ExportTest,MatchMethodTests,ResultSetTest,FakeDataStoreTest,HintDefinitionTests,TestAdministrativeFlows,GRRTempFileTestFilename,FilestoreStatsCronFlowTest,ForemanTests,ReportsTest,GRRBaseTest,TestEndToEndTestFlow,DictTest,TestArtifactCollectorsRealArtifacts,KnowledgeBaseUserMergeTest,HandlerTests,NetstatTest,BasicContextTests,ArtifactHandlingTest,ObjcTest,TestWindowsMemory,WindowsActionTests,GrepTest,SystemCronFlowTest,UtilsTest,DataStoreCSVBenchmarks,CheckLoaderTests,TDBDataStoreCSVBenchmarks,ArtifactKBTest,MultiShardedQueueManagerTest,ConfigActionTest,SearchTest,RegistryVFSTests,OSXDriverTests,RekallTestSuite,GeneralFlowsTest,FlowStateTest,TestCryptoTypeInfos,AFF4Benchmark,GRRArtifactTest,GRRFuseDatastoreOnlyTest,TestOSXFileParsing,AFF4SymlinkTest,FlowTestsBaseclass,ConditionTest,RekallVADParserTest
```

Comment:
These are worthy goals and I agree with them. It does not appear that this CL achieves those goals:

1. Do you know the resource limitation? we are turning off most of the tests - maybe we have just enough resources to run helloworld.py and no more? 

Note that I was not suggesting running more that one test concurrently (as it is done by the multiprocessing mode) but running them in their own process each. This should have the same memory requirements as running them in their own process with the advantage that you have a new global state each time (more manageable for tests).

2. What course of action will be required to get the tests to pass? Why not just do it now? It is still not clear by this cl because we dont really know why the tests fail since you turned them off.

3. We can illustrate the way tests are supposed to run by documenting it - on a normal machine with normal amount of ram the tests pass fine. It seems to me that you are arguing that we should change the entire test mechanism to fit a memory constraint setup which is specially only applicable to travis. This decision can not be taken lightly - we must consider what we are giving up (nice formatting of test output, multiprocessing support etc). 

If one has to run the tests manually anyway for a PR since most tests are switched off then automation is not an issue here. At this stage we dont have enough PRs coming in that we would benefit in automatically turning away a PR without review just because it breaks some tests.

Your link is showing that some tests are failing due to installation problems. Its also very hard to read because it includes debug output of all the tests - even passing ones. It is not useful at this stage since if a PR comes in that breaks something it would be very hard for someone to find exactly what was broken and why.

BTW there are other options than travis that we should explore - if travis is simply not suitable for running our tests we should not settle for it. Is it possible to get a larger memory allocation on  travis - that's also another option.